rfcs/006-package-management.md

```diff
@@ -0,0 +1,243 @@
+---
+feature: package management
+start-date: 2024-06-28
+author: Joe Neeman
+---
+
+# Package management
+
+People want to reuse code, but Nickel doesn't currently have a good way to do
+it. We should have a way to fetch packages and make them easily available to
+Nickel code. The mechanism needs to be predictable (it should fetch the code
+that the user expects to fetch) and reliable (if it works on my machine it
+should work on your machine).
+
+## The manifest file
+
+We will require a manifest file in order to import packages. Manifest files must
+be named `package.ncl`, and they are found by searching up from the file being
```

Comment:
I wonder if we should accept `package.toml`, `.json`, etc. Maybe it would be good hygiene to not have arbitrary Nickel code in here, as it could lead to some obfuscation. But `package` is so pervasive that we might try to find another word. We can just try to find a name for the package manager and use that instead.